adapters/ddf-adapter/src/main/java/com/connexta/replication/adapters/ddf/DdfNodeAdapter.java

```diff
@@ -261,13 +261,19 @@ public String getSystemName() {
 
   @Override
   public QueryResponse query(QueryRequest queryRequest) {
-    Iterable<Metadata> results =
-        ResultIterable.resultIterable(
-            this::getRecords,
-            new QueryRequestImpl(
-                createQueryFilter(queryRequest),
-                queryRequest.getStartIndex(),
-                queryRequest.getPageSize()));
+    Iterable<Metadata> results;
+    // Failed items and new items are replicated separately due to a bug in DDF that was only fixed
+    // recently
+    if (queryRequest.getFailedItemIds().isEmpty()) {
+      results =
+          ResultIterable.resultIterable(this::getRecords, createDdfQueryRequest(queryRequest));
+    } else {
+      results =
```

Comment:
if we made the createDdfFailedItemsQueryRequest just return null if there were no failed items and then had resultIterable just filter out null values it would simplify this. Not sure that it's worth it but just a thought.

Reply:
I made the changes you suggested.